[clang-tidy][readability-container-contains] Use hasOperands when possible

nicovank opened the pull request:
Simplifies two cases and matches two new cases previously missing, `container.end() [!=]= container.find(...)`.

I had split this change from #107521 for easier review.

EugeneZelenko commented on clang-tools-extra/docs/ReleaseNotes.rst:
```diff
@@ -167,7 +167,8 @@ Changes in existing checks
 - Improved :doc:`readability-container-contains
   <clang-tidy/checks/readability/container-contains>` check to let it work on
-  any class that has a ``contains`` method.
+  any class that has a ``contains`` method. Also now match previously missing
```
I don't think that second sentence is necessary. Original enhancement was introduced in same release.

nicovank committed:
[clang-tidy][readability-container-contains] Use hasOperands when possible

Simplifies two cases and matches two new cases previously missing, `container.end() [!=]= container.find(...)`.

I had split this change from #107521 for easier review.
